lib/workers/repository/init/inherited.ts

```diff
@@ -0,0 +1,102 @@
+import is from '@sindresorhus/is';
+import {
+  filterConfig,
+  mergeChildConfig,
+  removeGlobalConfig,
+} from '../../../config';
+import { parseFileConfig } from '../../../config/parse';
+import type { RenovateConfig } from '../../../config/types';
+import { validateConfig } from '../../../config/validation';
+import {
+  CONFIG_INHERIT_NOT_FOUND,
+  CONFIG_INHERIT_PARSE_ERROR,
+  CONFIG_VALIDATION,
+} from '../../../constants/error-messages';
+import { logger } from '../../../logger';
+import { platform } from '../../../modules/platform';
+import * as template from '../../../util/template';
+
+export async function mergeInheritedConfig(
+  config: RenovateConfig,
+): Promise<RenovateConfig> {
+  // typescript doesn't know that repo is defined
+  if (!config.repository || !config.inheritConfig) {
+    return config;
+  }
+  if (
+    !is.string(config.inheritConfigRepoName) ||
+    !is.string(config.inheritConfigFileName)
+  ) {
+    // Config validation should prevent this error
+    logger.error(
+      {
+        inheritConfigRepoName: config.inheritConfigRepoName,
+        inheritConfigFileName: config.inheritConfigFileName,
+      },
+      'Invalid inherited config.',
+    );
+    return config;
+  }
+  const repoParts = config.repository.split('/');
+  const templateConfig = {
+    topLevelOrg: repoParts.shift(),
+    parentOrg: repoParts.join('/'),
+    repostiory: config.repository,
```

Comment:
```suggestion
    repository: config.repository,
```

We probably should create a utility function in platforms to handle this. 
For sure we do something similar in multiple places